AppSettings/DevSettings/DevSettingLoader.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Mash.AppSettings.DevSettings
+{
+    public class DevSettingLoader : ISettingLoader
+    {
+        public string DevSettingFile { get; private set; }
+        private Dictionary<string, dynamic> _devSettings = new Dictionary<string, dynamic>();
+        private Dictionary<string, string> _connectionStrings = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Creates an instance of DevSettingLoader
+        /// </summary>
+        /// <param name="dir">The path to files holding developer-specific settings, defaults to current directory</param>
+        /// <param name="file">The developer's file to load, defaults to %username%.json</param>
+        public DevSettingLoader(string dir = null, string file = null)
+        {
+            if (String.IsNullOrEmpty(dir))
+            {
+                dir = Environment.CurrentDirectory;
+            }
+
+            string fileType = ".json";
+            if (String.IsNullOrWhiteSpace(file))
+            {
+                file = $"{Environment.UserName}{fileType}";
+            }
+
+            if (!file.EndsWith(fileType))
+            {
+                throw new ArgumentException("Only JSON files are valid.", nameof(file));
+            }
+
+            DevSettingFile = Path.Combine(dir, file);
+
+            string fileContents = File.ReadAllText(DevSettingFile);
+
+            LoadSettings(fileContents);
+        }
+
+        /// <summary>
+        /// Creates an instance of DevSetingLoader
+        /// </summary>
+        /// <param name="json">The json holding developer-specific settings</param>
+        public DevSettingLoader(string json)
+        {
+            LoadSettings(json);
+        }
+
+        private void LoadSettings(string json)
+        {
+            if (String.IsNullOrWhiteSpace(json))
+            {
+                return;
```

Comment:
should we just return ? no exceptions? because this is obviously a problem right?